compiler/src/lnntoamm/util.ts

```diff
@@ -2,7 +2,9 @@ import { v4 as uuid } from 'uuid';
 import Fn from './Fn';
 import Operator from './Operator';
 
-export const genName = () => '_' + uuid().replace(/-/g, '_');
+let counter = 0;
+export const genName = (name?: string) =>
+  name ? `${name}-n${counter++}` : '_' + uuid().replace(/-/g, '_');
```

Comment:
Why is both a counter an a UUID necessary for uniqueness?

Reply:
if `name` isn't `undefined`, then that's because it needs to be easier to read. if it is `undefined`, then i use uuid to make easier-to-read names and uuid names easier to separate and optically parse